### DISCUSSION TITLE
How to get rid of WARNING: Consider using the built-in warning_level option instead of using "-Wall".

### DISCUSSION THREAD

**Discussion by ilg-ul** [mesonbuild/meson#9909] [Q&A]
I'm getting this warning, but I could not find a way to set the `warning_level`.

My project builds multi-platform tests, and I cannot add it to the top `project()` since I need to enable it only for some platforms.

Is there any way to set it programatically? Or via an entry in the `cross.ini`, which, in my case, is defined for each platform.

**Comment by eli-schwartz** [comment]
> Or via an entry in the `cross.ini`, which, in my case, is defined for each platform.

Yes, you can: https://mesonbuild.com/Machine-files.html#meson-builtin-options
```ini
[built-in options]
warning_level = '0'
```

**Reply by ilg-ul** [reply]
My configuration used exactly `-Wall -Wextra`. In the Core options table I saw that the `warning_level ` values can be [0-3]; I tried them, but none of them brings exactly what I need, they also bring some other settings that I currently do not use.

Is there any trick to set these options selectively? Or to disable the warning and allow me to set the options manually?

**Reply by andy5995** [reply]
@ilg-ul I saw 'werror = false' in one example from the docs, maybe if you set warning level to 0 and add

>wall = true
wextra = true

maybe that would do what you want?

**Reply by ilg-ul** [reply]
Maybe I did something wrong, but both wall and wextra seem to be ignored.

**Reply by rusty-snake** [reply]
`werror` is a builtin-option in meson, while `wall` and `wextra` aren't.
https://mesonbuild.com/Builtin-options.html

**Reply by ravisynapse** [reply]
>@eli-schwartz @tabedzki @rusty-snake Can we explicitly add wall in in meson build file ? I heard that -Wall is default enabled in Meson build system .arm-ilo7-linux-gnueabi-gcc is the compiler. Any flags passed to the compiler should appear explicitly on command line. But don't see "-Wall".
> Generate build by explicitly passing "-Wall" to gcc. Compare the build log w/ and w/o passing "-Wall" explicitly.
> I am facing compiler parse error when I pass -Wall option

**Reply by andy5995** [reply]
> > @eli-schwartz @tabedzki @rusty-snake Can we explicitly add wall in in meson build file ? I heard that -Wall is default enabled in Meson build system .arm-ilo7-linux-gnueabi-gcc is the compiler. Any flags passed to the compiler should appear explicitly on command line. But don't see "-Wall".
> > Generate build by explicitly passing "-Wall" to gcc. Compare the build log w/ and w/o passing "-Wall" explicitly.
> > I am facing compiler parse error when I pass -Wall option

@ravisynapse Here's an example of how to explicitly add a flag is in https://github.com/theimpossibleastronaut/rmw/blob/998a405d4325385113eb2d2966581d658c74e1b4/meson.build#L9-L26

**Reply by ravisynapse** [reply]
@andy5995  .From my experiment found that -Wall is not enabled in meson .As per the meson document says -Wall is default enabled in meson build system .But I don't see it during the compile time .
Can you please help me to enable -Wall in meson ?

Thank you ,
Ravi

**Reply by andy5995** [reply]
@ravisynapse If you don't see '-Wall' during compile time, in your build directory, run `ninja clean` then `ninja -v`.

if no 'warning_level' is specified as a keyword arg to 'default_options', [then 'warning_level' defaults to 1](https://mesonbuild.com/Builtin-options.html#core-options), which will add the '-Wall' option.

```
project(
  'rmw',
  'c',
  version: '0.8.1.555',
  meson_version : '>= 0.56.0',
  default_options: ['warning_level=3']
  )
```

Please zip the 'compile_commands.json' file in your build directory and attach it to your reply.

Also copy and paste in your 'meson.build'.

**Reply by andy5995** [reply]
> yocto framework with bitbake

@ravisynapse I have never heard of those before and don't know anything about them. And you didn't attach the compile_commands.json file like I asked you too.
